internal/static/testdata/pkg-with-errors/app01/app.go

```diff
@@ -0,0 +1,22 @@
+package app01
+
+import "github.com/dogmatiq/dogma"
+
+const (
+	// AppKey is the application key.
+	AppKey = "7c3c67dd-6c0b-4952-97d5-54c75fc7a1c6"
+)
+
+// Below is the deliberate illegal Go syntax to test loading of the packages
+// with errors.
+This is deliberate illegal syntax
```

Comment:
Just to confirm, this is to cover the path where `ssautil.AllPackages()` returns a `nil` package?

Bonus question, after using does `prog.AllPackages()` include these `nil` packages too?

Reply:
> Just to confirm, this is to cover the path where ssautil.AllPackages() returns a nil package?

Yep, that's to cover that scenario which is now in a separate directory `testdata/apps/invalid-syntax-app`.

> Bonus question, after using does prog.AllPackages() include these nil packages too?

It appears that `prog.AllPackages()` only returns *loaded* packages. Since the one with the error could not be parsed and loaded, `prog.AllPackages()` does not return it in this particular test where the package with the invalid syntax is introduced. One of the interesting thing though is that `prog.AllPackages()` returns even internal packages of all imports. Below is the list of all packages that  `prog.AllPackages()` returns during the test with invalid syntax:
"internal/unsafeheader"
- "internal/reflectlite"
- "internal/race"
- "internal/oserror"
- "time"
- "strconv"
- "reflect"
- "os"
- "internal/bytealg"
- "math"
- "math/bits"
- "sort"
- "internal/syscall/execenv"
- "runtime/internal/sys"
- "runtime"
- "sync"
- "syscall"
- "context"
- "unicode/utf8"
- "io"
- "internal/syscall/unix"
- "sync/atomic"
- "internal/poll"
- "unsafe"
- "internal/cpu"
- "runtime/internal/atomic"
- "runtime/internal/math"
- "fmt"
- "errors"
- "unicode"
- "internal/fmtsort"
- "internal/testlog"
- "github.com/dogmatiq/dogma"

Reply:
Ok, that's interesting. I was wondering if we could use `prog.AllPackages()` and avoid having to skip the `nil`, but it looks like it needs to stay the way it is. No problem!